api/src/main/java/marquez/api/SearchResource.java

```diff
@@ -72,6 +83,48 @@ public Response search(
     return Response.ok(new SearchResults(searchResults)).build();
   }
 
+  @Timed
+  @ResponseMetered
+  @ExceptionMetered
+  @GET
+  @Produces(APPLICATION_JSON)
+  @Path("/jobs")
+  public Response searchJobs(@QueryParam("q") @NotBlank String query) throws IOException {
+    return formatEsResponse(this.searchService.searchJobs(query));
+  }
+
+  @Timed
+  @ResponseMetered
+  @ExceptionMetered
+  @GET
+  @Produces(APPLICATION_JSON)
+  @Path("/datasets")
+  public Response searchDatasets(@QueryParam("q") @NotBlank String query) throws IOException {
+    return formatEsResponse(this.searchService.searchDatasets(query));
+  }
+
+  private Response formatEsResponse(SearchResponse<ObjectNode> response) {
+    List<ObjectNode> hits =
+        response.hits().hits().stream().map(Hit::source).collect(Collectors.toList());
+    List<Map<String, List<String>>> highlights =
+        response.hits().hits().stream().map(Hit::highlight).collect(Collectors.toList());
+
+    return Response.ok(new EsResult(hits, highlights)).build();
+  }
+
+  @ToString
+  public static final class EsResult {
+    @Getter private final List<ObjectNode> hits;
```

Comment:
For now, this payload would work, but we would need to generalize search results:

```
{
  "results": [Dataset,Job]
}
```